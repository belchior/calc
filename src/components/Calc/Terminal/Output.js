import React, {Component} from 'react';
import {connect} from 'react-redux';

class Output extends Component {
  componentDidUpdate() {
    this.outputElem.scrollTop = this.outputElem.scrollHeight;
  }

  render() {
    const results = this.props.results || [];
    return (
      <div className="output" ref={(output) => this.outputElem = output}>
        {results.map((item, index) => {
          return <output className="result" onClick={this.props.onClick} key={index}>{item}</output>;
        })}
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return state.terminal;
};

const mapDispatchToProps = (dispatch) => {
  return {
    onClick: (e) => {
      dispatch({type: 'NUMBER', calc: 'terminal', payload: e.target.textContent});
    }
  };
};

export {Output};
export default connect(mapStateToProps, mapDispatchToProps)(Output);
