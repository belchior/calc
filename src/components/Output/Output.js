import React, {Component} from 'react';

export class Output extends Component {
  componentDidUpdate() {
    this.outputElem.scrollTop = this.outputElem.scrollHeight;
  }

  render() {
    const results = this.props.results || [];
    return (
      <div className="output" ref={(output) => this.outputElem = output}>
        {results.map((item, index) => {
          return <output className="result" onClick={this.props.onClick} key={index}>{item}</output>;
        })}
      </div>
    );
  }
}

export default Output;
