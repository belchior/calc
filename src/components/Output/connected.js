import { connect } from 'react-redux';
import Output from './Output';

export const mapStateToProps = (state) => {
  return state.terminal;
};

export const mapDispatchToProps = (dispatch) => {
  return {
    onClick: (e) => {
      dispatch({ type: 'NUMBER', calc: 'termsinal', payload: e.target.textContent });
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Output);
