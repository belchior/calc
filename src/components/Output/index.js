import Output, { mapStateToProps, mapDispatchToProps } from './connected';

export { mapStateToProps, mapDispatchToProps, };
export default Output;
