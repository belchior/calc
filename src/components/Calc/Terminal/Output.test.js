import React from 'react';
import {Output} from './Output';
import renderer from 'react-test-renderer';

it('should render Output with 0 items without crashing', () => {
  let tree = renderer.create(<Output />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render Output with 2 items without crashing', () => {
  let tree = renderer.create(<Output results={['1.2', '3.4']}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
