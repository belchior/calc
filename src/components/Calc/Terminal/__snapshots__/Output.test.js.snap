// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Output with 0 items without crashing 1`] = `
<div
  className="output"
/>
`;

exports[`should render Output with 2 items without crashing 1`] = `
<div
  className="output"
>
  <output
    className="result"
    onClick={undefined}
  >
    1.2
  </output>
  <output
    className="result"
    onClick={undefined}
  >
    3.4
  </output>
</div>
`;
